x/cfevesting/types/message_move_available_vesting.go

```diff
@@ -38,13 +38,22 @@ func (msg *MsgMoveAvailableVesting) GetSignBytes() []byte {
 }
 
 func (msg *MsgMoveAvailableVesting) ValidateBasic() error {
-	_, err := sdk.AccAddressFromBech32(msg.FromAddress)
+	_, _, err := ValidateMsgMoveAvailableVesting(msg.FromAddress, msg.ToAddress)
 	if err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid fromAddress address (%s)", err)
+		return err
 	}
-	_, err = sdk.AccAddressFromBech32(msg.ToAddress)
+	return nil
+}
+
+func ValidateMsgMoveAvailableVesting(fromAddress string, toAddress string) (fromAccAddress sdk.AccAddress, toAccAddress sdk.AccAddress, error error) {
```

Comment:
create validation hierarchy for move availeble, move available by denoms, adn split vesting